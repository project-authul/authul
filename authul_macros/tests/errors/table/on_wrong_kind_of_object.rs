#[authul_macros::return_as_is]
enum Blargh {}

#[authul_macros::authul_table]
enum Foo {}

#[authul_macros::authul_table]
fn foo() {}

#[authul_macros::authul_table]
struct Bar;

#[authul_macros::authul_table]
struct Baz(String);

fn main() {
}
