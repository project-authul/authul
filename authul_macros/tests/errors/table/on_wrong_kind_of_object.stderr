error: authul_table only applies to structs
 --> tests/errors/table/on_wrong_kind_of_object.rs:5:1
  |
5 | enum Foo {}
  | ^^^^

error: authul_table only applies to structs
 --> tests/errors/table/on_wrong_kind_of_object.rs:8:1
  |
8 | fn foo() {}
  | ^^

error: authul_table only works on structs with named fields
  --> tests/errors/table/on_wrong_kind_of_object.rs:11:11
   |
11 | struct Bar;
   |           ^

error: authul_table only works on structs with named fields
  --> tests/errors/table/on_wrong_kind_of_object.rs:14:11
   |
14 | struct Baz(String);
   |           ^^^^^^^^
